db/migrate/20221020131636_add_user_ref_to_posts.rb

```diff
@@ -0,0 +1,5 @@
+class AddUserRefToPosts < ActiveRecord::Migration[7.0]
+  def change
+    add_reference :posts, :user, null: false, foreign_key: true
```

Comment:
- [ ] The post should have an `author_id` column, not `user_id`. And so these references should be `author`, not `user`. You can point the foreign key to the user table. You can look at this link for hint: [https://stackoverflow.com/a/42056089](https://stackoverflow.com/a/42056089)